quickwit/quickwit-serve/src/lib.rs

```diff
@@ -296,6 +298,9 @@ pub async fn serve_quickwit(
                 .stack_add_source_layer(broker_layer.clone())
                 .stack_delete_source_layer(broker_layer.clone())
                 .stack_toggle_source_layer(broker_layer)
+                .stack_layer(tower::limit::GlobalConcurrencyLimitLayer::new(
+                    MAX_CONCURRENT_METASTORE_REQUESTS,
```

Comment:
This should be configurable, at the minimum, with an env var. I also don't think that's an appropriate value for a PostgreSQL metastore. Let's discuss this.

Reply:
added an env variable and move it to the client side.